sdks/go/pkg/beam/runners/dataflow/dataflow.go

```diff
@@ -90,15 +94,20 @@ func Execute(ctx context.Context, p *beam.Pipeline) error {
 	}
 
 	if *cpuProfiling != "" {
-		beam.PipelineOptions.Set("cpu_profiling", "true")
-		beam.PipelineOptions.Set("storage_path", "/var/opt/google/traces")
+		perf.EnableProfCaptureHook("gcs_profile_writer", []string{*cpuProfiling})
 	}
 
 	if *sessionRecording != "" {
-		beam.PipelineOptions.Set("session_recording", "true")
-		beam.PipelineOptions.Set("storage_path", "/var/opt/google/traces")
+		// TODO(wcn): implement this.
```

Comment:
Please open a JIRA.

Reply:
I actually did, I didn't fix the comment.